lgc/builder/YCbCrConvertHandler.h

```diff
@@ -0,0 +1,272 @@
+  // Generate sampler descriptor for chroma channel
+  void genSamplerDescChroma();
+
+  // Generate image descriptor for chroma channel
+  void genImgDescChroma();
+
+  // Convert from YCbCr to RGBA color space
+  llvm::Value *convertColorSpace();
+
+  // Sample YCbCr data from each plane
+  // Note: Should be called after genImgDescChroma and genSamplerDescChroma completes
+  void sampleYCbCrData();
+
+  // Set the ST coords
+  void setCoord(llvm::Value *coordS, llvm::Value *coordT);
+
+  // Set the sample infomation of Luma channel
+  inline void setSampleInfoLuma(llvm::Type *resultType) { m_resultType = resultType; }
+
+  // Set GfxIp version
+  inline void setGfxIpVersion(GfxIpVersion *gfxIp) { m_gfxIp = gfxIp; }
+
+  // Implement transfer from ST coordinates to UV coordiantes operation
+  llvm::Value *transferSTtoUVCoords(llvm::Value *coordST, llvm::Value *scale);
+
+  // Implement the adjustment of UV coordiantes when the sample location associated with
+  // downsampled chroma channels in the X/XY dimension occurs
+  llvm::Value *calculateImplicitChromaUV(ChromaLocation offset, llvm::Value *coordUV);
+
+  // Transfer IJ coordinates from UV coordinates
+  llvm::Value *transferUVtoIJCoords(SamplerFilter filter, llvm::Value *coordUV);
+
+  // Calculate UV offset to top-left pixel
+  llvm::Value *calculateUVoffset(llvm::Value *coordUV);
+
+  // Implement bilinear blending
+  llvm::Value *bilinearBlend(llvm::Value *alpha, llvm::Value *beta, llvm::Value *coordTL, llvm::Value *coordTR,
+                             llvm::Value *coordBL, llvm::Value *coordBR);
+
+  // Implement wrapped YCbCr sample
+  llvm::Value *wrappedSample(YCbCrWrappedSampleInfo &wrapInfo);
+
+  // Implement reconstructed YCbCr sample operation for downsampled chroma channels in the X dimension
+  llvm::Value *reconstructLinearXChromaSample(XChromaSampleInfo &xChromaInfo);
+
+  // Implement reconstructed YCbCr sample operation for downsampled chroma channels in both X and Y dimension
+  llvm::Value *reconstructLinearXYChromaSample(XYChromaSampleInfo &xyChromaInfo);
+
+  // Implement interanl image sample for YCbCr conversion
+  llvm::Value *createImageSampleInternal(llvm::SmallVectorImpl<llvm::Value *> &coords, YCbCrSampleInfo *ycbcrInfo);
+
+  // Generate sampler descriptor for YCbCr conversion
+  llvm::Value *generateSamplerDesc(llvm::Value *samplerDesc, SamplerFilter filter, bool forceExplicitReconstruction);
+
+  // Implement range expanding operation on checking whether the encoding uses full numerical range on chroma channel
+  llvm::Value *rangeExpand(SamplerYCbCrRange range, const uint32_t *channelBits, llvm::Value *sampleResult);
+
+  // Implement the color transfer operation for conversion from YCbCr to RGB color model
+  llvm::Value *convertColor(llvm::Type *resultTy, SamplerYCbCrModelConversion colorModel, SamplerYCbCrRange range,
+                            uint32_t *channelBits, llvm::Value *imageOp);
+
+private:
+  SamplerYCbCrConversionMetaData *m_metaData;
```

Comment:
You don't initialize all data members in the constructor. I know some members are for intermediate uses.
 For those not-handled members, add the initialization here, such as ` m_width = nullptr`. Also, please add blank line to group relevant members to improve readability, something like this:

```
  llvm::Value *m_chromaSampDesc;

  llvm::Value *m_coordS;
  llvm::Value *m_coordT;
  llvm::Value *m_coordU;
  llvm::Value *m_coordV;
  llvm::Value *m_coordI;
  llvm::Value *m_coordJ;

  GfxIpVersion *m_gfxIp;
```